frontend/packages/console-shared/src/hooks/useURLPolls.ts

```diff
@@ -0,0 +1,85 @@
+import * as React from 'react';
+import * as _ from 'lodash';
+import {
+  URLPollResult,
+  URLPollsResultMap,
+  UseURLPolls,
+} from '@console/dynamic-plugin-sdk/src/api/internal-types';
+import { useSafeFetch } from '@console/internal/components/utils';
+import usePolls from './usePolls';
+
+export const URL_POLL_DEFAULT_DELAY = 15000; // 15 seconds
+
+export type PollsData = {
+  callback: () => any;
+  delay?: number;
+  dependencies?: any[];
+};
+
+export type PollsDataMap = {
+  [key: string]: PollsData;
+};
+
+const useURLPolls: UseURLPolls = (pollDataMap) => {
+  const [pollsDataMap, setPollsDataMap] = React.useState<PollsDataMap>();
+  const [resultsMap, setResultsMap] = React.useState<URLPollsResultMap>(
+    Object.keys(pollDataMap)?.reduce((acc, url) => ({ ...acc, [url]: {} }), {}) || {},
+  );
+  const safeFetch = useSafeFetch();
+
+  // eslint-disable-next-line consistent-return
+  React.useEffect(() => {
```

Comment:
when the useEffect is retriggered, it should call `AbortController` of `useSafeFetch`